upgrade/upgrade-scripts/upgrade-container

```diff
@@ -527,8 +527,23 @@ function do_upgrade_container_not_in_place() {
 		die "failed to migrate configuration for '$CONTAINER'"
 }
 
+function get_type() {
+	local root origin
+
+	root=$(sudo zfs list -Hpo name /var/lib/machines/${CONTAINER})
+	[[ -n "${root}" ]] || die "Failed to obtain root filesystem for container '${CONTAINER}'"
+	origin=$(sudo zfs get -Hpo value origin ${root})
+
+	case "${origin}" in
+	-)
+		echo "not-in-place" ;;
+	*)
+		echo "in-place" ;;
+	esac
+}
+
 function do_upgrade_container() {
-	local type="$1"
+	local type=${get_type}
```

Comment:
IMO, it'd be more simple to just do this:
```
origin=$(sudo zfs get -Hpo value origin ${root})
case "$origin" in
-)
    do_upgrade_container_in_place || ...
    ;;
*)
    do_upgrade_container_not_in_place || ...
    ;;
esac
```

The `get_type` function is only used once, so it'd be a little more clear (to me, at least) if we just combined it with this `do_upgrade_container` function.

Reply:
Although, now I see you're exposing the `get_type` function via `upgrade-container get-type`...

if we think it'd be useful to allow callers to get the type of the container like you're doing, then I think what you have makes sense.